drivers/phoenixcontact_modbus.c

```diff
@@ -110,17 +110,17 @@ void upsdrv_initinfo(void)
 	/* upsh.instcmd = instcmd; */
 	/* upsh.setvar = setvar; */
 
-	mrir(modbus_ctx, 0x0005, 4, &tab_reg);
+	mrir(modbus_ctx, 0x0005, 4, tab_reg);
 
 	/*	Method provided from Phoenix Conatct to establish the UPS model:
 		Read registers from 0x0005 to 0x0008 and "concatenate" them with the order
 		0x0008 0x0007 0x0006 0x0005 in hex form, convert the obtained number from hex to dec.
 		The first 7 most significant digits of the number in dec form are the part number of
 		the UPS.*/
 
-	PartNumber = (tab_reg[3] * 65536) + tab_reg[2];
-	PartNumber = (PartNumber * 65536) + tab_reg[1];
-	PartNumber = (PartNumber * 65536) + tab_reg[0];
+	PartNumber = (tab_reg[3] << 16) + tab_reg[2];
```

Comment:
Before I merge, did you double-check on hardware that this part number magic works after the changes in the PR?
Namely, after a new look I worry if `tab_reg[3]` should be cast into `uint64_t` before bit-shifting, so it does not overflow as a 16-bit number on any system/compiler. None complained about smelling a fishy behavior in CI ranging from clang3 and gcc4 up to modern, though.